packages/annotorious-core/src/model/W3CAnnotation.ts

```diff
@@ -97,22 +113,58 @@ export const parseW3CBodies = (
     purpose,
     value,
     created,
-    creator: creator ? 
-      typeof creator === 'object' ? { ...creator }: creator :
+    creator: creator ?
+      typeof creator === 'object' ? { ...creator } : creator :
       undefined,
     ...rest
-  }
+  };
 
 });
 
 /** Serialization helper to remove core-specific fields from the annotation body **/
-export const serializeW3CBodies = (bodies: AnnotationBody[]): W3CAnnotationBody[] => 
+export const serializeW3CBodies = (bodies: AnnotationBody[]): W3CAnnotationBody[] =>
   bodies.map(b => {
     const w3c = { ...b };
     delete w3c.annotation;
 
     if (w3c.id?.startsWith('temp-'))
       delete w3c.id;
-  
+
     return w3c;
-  });
\ No newline at end of file
+  });
+
+export const parseW3CTarget = <A extends AnnotationTarget, T extends W3CAnnotationTarget<T['selector']> = W3CAnnotationTarget>(
+  target: T,
+  annotationId: string,
+  lifecycleInfo: AnnotationLifecycleInfo,
+  selector: A['selector']
+): A =>
+  ({
+    ...target,
+    ...lifecycleInfo,
+    annotation: annotationId,
+    selector,
+  } as unknown as A);
+
+export const serializeW3CTarget = <A extends AnnotationTarget, T extends W3CAnnotationTarget<T['selector']> = W3CAnnotationTarget>(target: A, source: string, selector: T['selector']): T => {
+  const { creator, created, updatedBy, updated, ...serializableRest } = target;
+  return {
+    ...serializableRest,
+    selector,
+    source
+  } as unknown as T;
+};
```

Comment:
Intentionally using the `as unknown as Something` because Typescript doesn't know that that `target` and `serializableRest` makes the expected type complete